### ISSUE TITLE
How to Generate Zone?

### ISSUE THREAD

**Issue by arraysize** [haxball/haxball-issues#1466]
How to check the ball is in a zone? When I moved the cursor to a point in Haxpuck, it gives to me X and Y coordinates. 

For example: https://i.imgur.com/8GfrYY2.jpg

**Comment by thenorthstar** [comment]
> How to check the ball is in a zone? When I moved the cursor to a point in Haxpuck, it gives to me X and Y coordinates.
> 
> For example: https://i.imgur.com/8GfrYY2.jpg

@arraysize

```js
var x1 = -200;
var x2 = 200;
var y1 = -200;
var y2 = 200;

function isBallInZone(){
    return (x1 < room.getDiscProperties(0).x && room.getDiscProperties(0).x < x2) && (y1 < room.getDiscProperties(0).y && room.getDiscProperties(0).y < y2);
}

function some_function(){
    if(isBallInZone()){ //Do something.}
    else{ //Do something.}
}

room.onGameTick = function(){
    isBallInZone();
}
```

**Comment by arraysize** [comment]
> > How to check the ball is in a zone? When I moved the cursor to a point in Haxpuck, it gives to me X and Y coordinates.
> > For example: https://i.imgur.com/8GfrYY2.jpg
> 
> @arraysize
> 
> ```js
> var x1 = -200;
> var x2 = 200;
> var y1 = -200;
> var y2 = 200;
> 
> function isBallInZone(){
>     return (x1 < room.getDiscProperties(0).x && room.getDiscProperties(0).x < x2) && (y1 < room.getDiscProperties(0).y && room.getDiscProperties(0).y < y2);
> }
> 
> function some_function(){
>     if(isBallInZone()){ //Do something.}
>     else{ //Do something.}
> }
> 
> room.onGameTick = function(){
>     isBallInZone();
> }
> ```

https://i.imgur.com/52V0iST.jpg
is it true?

**Comment by thenorthstar** [comment]
> > > How to check the ball is in a zone? When I moved the cursor to a point in Haxpuck, it gives to me X and Y coordinates.
> > > For example: https://i.imgur.com/8GfrYY2.jpg
> > 
> > 
> > @arraysize
> > ```js
> > var x1 = -200;
> > var x2 = 200;
> > var y1 = -200;
> > var y2 = 200;
> > 
> > function isBallInZone(){
> >     return (x1 < room.getDiscProperties(0).x && room.getDiscProperties(0).x < x2) && (y1 < room.getDiscProperties(0).y && room.getDiscProperties(0).y < y2);
> > }
> > 
> > function some_function(){
> >     if(isBallInZone()){ //Do something.}
> >     else{ //Do something.}
> > }
> > 
> > room.onGameTick = function(){
> >     isBallInZone();
> > }
> > ```
> 
> https://i.imgur.com/52V0iST.jpg
> is it true?

@arraysize No. What you shown are vertexes (**{x,y}** couples). And what I told is the following.

![image](https://user-images.githubusercontent.com/68077608/133122265-4544eaef-d702-4392-8d27-84fa2656c8d2.png)

**Comment by Elinous-Electra** [comment]
https://discord.gg/eDkwYjQQBy
